source/Detector/Amptek/AmptekSDD123Server.cpp

```diff
@@ -0,0 +1,522 @@
+		connect(syncPacketTimer_, SIGNAL(timeout()), this, SLOT(syncDatagramTimeout()));
+		if(overrideTimeout == -1)
+			syncPacketTimer_->start(currentSyncPacket_.timeout());
+		else
+			syncPacketTimer_->start(overrideTimeout);
+	}
+}
+
+void AmptekSDD123Server::fakeSendDatagram(AmptekSDD123Packet packet, int overrideTimeout)
+{
+	if(socketLocallyBusy_){
+		AMErrorMon::alert(this, AMPTEK_SERVER_ALERT_PACKET_QUEUE_BUSY, "PACKET QUEUE IS BUSY (fake send datagram)");
+		packetQueue_.append(packet);
+	}
+	else{
+		socketLocallyBusy_ = true;
+		currentRequestPacket_ = packet;
+		currentRequestTime_.restart();
+		//udpDetectorSocket_->writeDatagram(packet.datagram(), packet.datagram().size(), configurationMap_->detectorAddress(), 10001);
+		connect(requestPacketTimer_, SIGNAL(timeout()), this, SLOT(requestDatagramTimeout()));
+		if(overrideTimeout == -1)
+			requestPacketTimer_->start(currentRequestPacket_.timeout());
+		else
+			requestPacketTimer_->start(overrideTimeout);
+	}
+}
+
+void AmptekSDD123Server::processPendingDatagrams()
+{
+	bool emitResponsePacketReady = false;
+
+	while(udpDetectorSocket_->hasPendingDatagrams()){
+		QByteArray datagram;
+		bool ok;
+		datagram.resize(udpDetectorSocket_->pendingDatagramSize());
+		QHostAddress *sourceAddress = new QHostAddress();
+		udpDetectorSocket_->readDatagram(datagram.data(), datagram.size(), sourceAddress);
+
+		if(datagram.size() < 520 && datagram.mid(0,2) == QByteArray::fromHex("f5fa")){
+			emitResponsePacketReady = true;
+
+			totalResponseDatagramSize_ = datagram.size();
+			totalResponseDatagram_.clear();
+			totalResponseDatagram_.append(datagram);
+			stillReceivingResponseDatagram_ = false;
+		}
+		else if(stillReceivingResponseDatagram_){
+			totalResponseDatagram_.append(datagram);
+			if(totalResponseDatagram_.size() == totalResponseDatagramSize_){
+				emitResponsePacketReady = true;
+				stillReceivingResponseDatagram_ = false;
+			}
+		}
+		else if(datagram.size() == 520){
+			if(datagram.mid(0,2) == QByteArray::fromHex("f5fa")){
+				stillReceivingResponseDatagram_ = true;
+				totalResponseDatagramSize_ = datagram.mid(4,2).toHex().toInt(&ok, 16) + 8;//+8 for header, command, length, and checksum
+				totalResponseDatagram_.clear();
+				totalResponseDatagram_.append(datagram);
+			}
+		}
+		delete sourceAddress;
+	}
+
+	if(emitResponsePacketReady)
+		emit responsePacketReady();
+}
+
+void AmptekSDD123Server::onResponsePacketReady()
+{
+	requestPacketTimer_->stop();
+	syncPacketTimer_->stop();
+
+	AmptekSDD123Packet lastRequestPacket = currentRequestPacket_;
+	AmptekSDD123Packet responsePacket(currentRequestPacket_.packetID(), totalResponseDatagram_);
+
+	processLocalStoredPacket(responsePacket);
+	processResponseFeedback(responsePacket);
+	processLastRequestPacket(lastRequestPacket);
+	sendPacketInQueue();
+}
+
+void AmptekSDD123Server::syncRequest(int timeout)
+{
+	QString hashTag = QString("%1").arg(--timeOutIDCounter_);
```

Comment:
Readability change here and in fakeSyncRequest.  I know it's being a bit overly pedantic, but I think the ease of reading it in the future will pay off with more than saving a couple lines of code.